tests/test_simulator.py

```diff
@@ -0,0 +1,67 @@
+import numpy
+
+from model import state, action
+import simulator
+
+
+class SingleDimStateSpace(state.StateSpace):
+    def __init__(self, len):
+        self.d = [1, len]
+
+    def dim(self):
+        return self.d
+
+    def states(self):
+        return [(0, i) for i in xrange(self.dim()[1])]
+
+
+class SingleDimActionSpace(action.ActionSpace):
+    def __init__(self, choices, valuations, len):
+        self.choices = choices
+        self.valuations = valuations
+        self.len = len
+
+    def get_actions(self, state):
+        for choice in self.choices:
+            potential_action = choice + state[1]
+            if 0 <= potential_action < self.len:
+                yield (choice, -1 * choice * choice + self.valuations[state])
+
+    def transition_probabilities(self, action, state):
+        loc = state[1]
+        for delta, prob in [(-1, .5), (0, .25), (1, .25)]:
+            next_loc = min(max(loc + action + delta, 0), self.len - 1)
+            if 0 <= next_loc < self.len:
+                yield (0, next_loc), prob
+
+
+# TODO: make fixture or abandon py.test
```

Comment:
I think I'll just stick with py.test.  I think the debug messages for unittest are harder to read